pages/api/v1/confirm.ts

```diff
@@ -0,0 +1,166 @@
+import { createClient } from "@supabase/supabase-js";
+import { Redis } from "@upstash/redis";
+import { NextRequest } from "next/server";
+import { z } from "zod";
+import { OrgJoinIsPaid } from "../../../lib/types";
+import { isValidBody } from "../../../lib/utils";
+import {
+  httpRequestFromAction,
+  processAPIoutput,
+} from "../../../lib/edge-runtime/requests.ts";
+
+export const config = {
+  runtime: "edge",
+};
+
+const OptionalStringZod = z.optional(z.string());
+
+const ConfirmZod = z.object({
+  conversation_id: z.number(),
+  user_api_key: OptionalStringZod,
+  org_id: z.number(),
+  button_clicked: z.enum(["yes", "no"]),
+});
+
+type ConfirmType = z.infer<typeof ConfirmZod>;
+
+// TODO: is this the best thing for open source? Would require users to have a redis account
```

Comment:
Good question - ideally the answer is no and if there's no upstash keys given we fall back on using the database (since all the messages are already being stored there, we can parse and run the fetch from the last message with the same convo id)

We want HN visitors to be able to set up and play locally with minimal setup

Tell me if you think this change will take a long time and we can reconsider